### DISCUSSION TITLE
Feature request: empty collection in `SampledFrom`

### DISCUSSION THREAD

**Discussion by MilesCranmer** [Seelengrab/Supposition.jl#48] [Design & Ideas]
I was wondering if empty collections could work in a `OneOf`, like the following?

```julia
julia> g = Data.SampledFrom(1:5) | Data.SampledFrom(1:0);

julia> example(g, 100)
ERROR: ArgumentError: `minimum` must be `<= maximum`!
```

In DynamicExpressions sometimes there are cases with 0 binary operators, but 1 unary operator. In which case it would be nice if `OneOf` simply skip over the binary nodes automatically since they aren't possible.

**Comment by Seelengrab** [comment]
The problem is that `SampledFrom` needs to produce some value, and it can't produce a value from an empty collection. It's the same problem you get with `reduce(*, [])` - what should happen in the empty case? There just isn't a value that can be returned here :shrug: For `reduce`, you can pass `init` but that doesn't really make sense for `SampledFrom` (and also wouldn't be what you're looking for, since it's not "skipping").

**Reply by MilesCranmer** [reply]
That makes sense. I guess I'm wondering if a `SampledFrom` could be marked as "empty", and `OneOf` simply skip over it or something. But perhaps this is a bit of an edge case and not worth a general solution, I should just work around it.

**Reply by Seelengrab** [reply]
Yes, exactly. It's a bit of a philosophical point; in general, when the API requires some decision about how an edge case is handled I prefer to punt that on the user instead of trying to pick something arbitrary. Every such edge case that can come up is indicative of an edge case in the program that will ultimately be tested with the resulting `Possibility`, and as such it needs a decision in the context of that program.

After all, part of the point of Supposition.jl is exactly finding such unhandled edge cases in some program - if that already happens at the stage of modeling the input, that's a win in my book!
